cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2600,6 +2602,7 @@ endmacro()
 if(ARROW_WITH_BZ2)
   resolve_dependency(BZip2 PC_PACKAGE_NAMES bzip2)
 
+  # XXX if resolve_dependency is called, won't we have either BZip2::BZip2 or an error?
   if(NOT TARGET BZip2::BZip2)
     add_library(BZip2::BZip2 UNKNOWN IMPORTED)
     set_target_properties(BZip2::BZip2
```

Comment:
Is this dead code?

Reply:
This block is for CMake < 3.12. So we can remove this.